src/hotspot/share/gc/shenandoah/shenandoah_globals.hpp

```diff
@@ -300,17 +293,20 @@
           "failures, which will trigger stop-the-world Full GC passes.")    \
           range(1.0,100.0)                                                  \
                                                                             \
-  product(double, ShenandoahGenerationalEvacWaste, 2.0, EXPERIMENTAL,       \
-          "For generational mode, how much waste evacuations produce "      \
-          "within the reserved space.  Larger values make evacuations "     \
-          "more resilient against evacuation conflicts, at expense of "     \
-          "evacuating less on each GC cycle.  Smaller values increase "     \
-          "the risk of evacuation failures, which will trigger "            \
-          "stop-the-world Full GC passes.  The default value for "          \
-          "generational mode is 2.0.  The reason for the higher default "   \
-          "value in generational mode is because generational mode "        \
-          "enforces the evacuation budget, triggering degenerated GC "      \
-          "which upgrades to full GC whenever the budget is exceeded.")     \
+  product(double, ShenandoahOldEvacWaste, 1.6, EXPERIMENTAL,                \
+          "How much waste evacuations produce within the reserved space. "  \
+          "Larger values make evacuations more resilient against "          \
+          "evacuation conflicts, at expense of evacuating less on each "    \
+          "GC cycle.  Smaller values increase the risk of evacuation "      \
+          "failures, which will trigger stop-the-world Full GC passes.")    \
+          range(1.0,100.0)                                                  \
+                                                                            \
+  product(double, ShenandoahPromoEvacWaste, 1.2, EXPERIMENTAL,              \
+          "How much waste evacuations produce within the reserved space. "  \
+          "Larger values make evacuations more resilient against "          \
+          "evacuation conflicts, at expense of evacuating less on each "    \
+          "GC cycle.  Smaller values increase the risk of evacuation "      \
+          "failures, which will trigger stop-the-world Full GC passes.")    \
```

Comment:
The terms `waste` and `conflicts` are not clear to me. Can we write this in a way that someone not familiar with the internals of the collector can understand without having to dig through the code to understand what `waste` and `evacuation conflict` are? Or include just enough for them to be able to use these knobs with some amount of instinct rather than knob-twiddling "because it was there".

It looks to me like a "magic number estimate" that the user produces out of thin air that the collector then uses in its accounting, and then real life unfolds in such a manner that this determines risk vs reward or efficiency of space usage. In that sense this is more like a "magic padding factor". If so, I'd describe in a line or two what makes the actual space usage more than the size of the evacuated objects, why it's difficult to accurately assess ahead of time, and why a larger value potentially wastes space by allowing more headroom while avoiding running out of space while a smaller value wastes less space while running a greater risk of running out of space in tight situations where it attempted an evacuation.

The talk of waste as related to "conflict" is a bit confusing to me at least without knowing what conflict is and why it must lead to wastage of space.

This probably needs more discussion to see why we may not be able to learn something better based on measurement as the collections unfold, based on an initial rough estimate to begin with and then refining it with a suitable padding for uncertainty over time.